File: api/Ganymede.Api.Data/Campaign.cs
using Ganymede.Api.Data.Monsters;
using Ganymede.Api.Data.Rulesets;
using Ganymede.Api.Data.Spells;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Ganymede.Api.Data
{
    public class Campaign
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }


        public int RulesetID { get; set; }
        [ForeignKey("RulesetID")]
        public Ruleset Ruleset { get; set; }

        public string AppUserId { get; set; }
        [ForeignKey("AppUserId")]
        public AppUser User { get; set; }

        public ICollection<Monster> Monsters { get; set; }
        public ICollection<Spell> Spells { get; set; }
    }
}

File: src/app/models/core/campaign.ts
import { Ruleset } from "./rulesets/ruleset";
import { Spell } from "./spells/spell";
import { AppUser } from "./appUser";

export class Campaign {
	public id: number;
	public name: number;
	public description: number;

	public rulesetID: number;
	public ruleset: Ruleset;

	public appUserID: string;
	public user: AppUser;

	public monsters: any[];
	public spells: Spell[];

	public static getDefault(): Campaign {
		const campaign = new Campaign();
		campaign.id = -1;

		return campaign;
	}

	public static isEqual(a: Campaign, b: Campaign): boolean {
		return (
			a.id === b.id &&
			a.name === b.name &&
			a.description === b.description &&
			a.rulesetID === b.rulesetID &&
			a.appUserID === b.appUserID
		);
	}
}
